codesign/lib/src/google_cloud_storage.dart

```diff
@@ -15,19 +15,27 @@ class GoogleCloudStorage {
     required this.processManager,
     required this.rootDirectory,
     required this.commitHash,
+    required this.optionalSwitch,
   });
 
   final ProcessManager processManager;
   final Directory rootDirectory;
   final String commitHash;
-  final String gsCloudBaseUrl = 'gs://flutter_infra_release';
+  final List<String> optionalSwitch;
+  final String bucketPrefix = 'gs://flutter_infra_release';
 
   /// Method to upload code signed flutter engine artifact to google cloud bucket.
   Future<void> uploadEngineArtifact({
     required String from,
     required String destination,
   }) async {
-    final String destinationUrl = '$gsCloudBaseUrl/flutter/$commitHash/$destination';
+    final String destinationUrl;
+    if (optionalSwitch.isEmpty) {
+      destinationUrl = '$bucketPrefix/flutter/$commitHash/$destination';
+    } else {
+      final String rawName = destination.replaceAll(".zip", "");
```

Comment:
Why does this not need the zip but the other one does?  Can you add a comment?

Reply:
Sure, I added a comment. The trimming of .zip is mostly used to create the full gcloud bucket path for ios usb artifacts. For example, assume we have destination as ios-deploy.zip. If this is an engine artifact, the bucket path would be something like ...blah/ios-deploy.zip, where the name is only used once. If this is an ios usb artifact, we process the bucket path to be something like ...blah/ios-deploy/...blah/ios-deploy.zip, which is sort of repeated in the gcloud bucket path.